src/client/render/secondstage.cpp

```diff
@@ -87,16 +87,20 @@ RenderStep *addPostProcessing(RenderPipeline *pipeline, RenderStep *previousStep
 	auto driver = client->getSceneManager()->getVideoDriver();
 
 	// configure texture formats
-	video::ECOLOR_FORMAT color_format = video::ECF_A8R8G8B8;
+	auto color_format = video::ECF_A8R8G8B8;
 	if (driver->queryTextureFormat(video::ECF_A16B16G16R16F))
 		color_format = video::ECF_A16B16G16R16F;
 
-	video::ECOLOR_FORMAT depth_format = video::ECF_D16; // fallback depth format
-	if (driver->queryTextureFormat(video::ECF_D32))
-		depth_format = video::ECF_D32;
+	auto depth_format = video::ECF_D16;
+	if (driver->queryTextureFormat(video::ECF_D24))
+		depth_format = video::ECF_D24;
```

Comment:
For preferring a 24-bit depth buffer over a 32-bit buffer:

- Do you have data how much it improves performance / have you observed a performance improvement?

- Looking through the issue tracker, it seems there have been some issues with z-fighting in the past. Have you tested this to make sure there are no new problems?

Reply:
I have done no testing beyond checking that world rendering generally works. No performance data either.
This is all just going off "24-bit seems to be the default in GL so that should be enough for us".

If x2048 was still here we could ask him why he preferred a 32-bit depth buffer but :shrug: 

> Looking through the issue tracker, it seems there have been some issues with z-fighting in the past

links please

Reply:
OK, looking into this further, the non-post-processing version of the pipeline has always used a 24-bit depth buffer, so this should be fine.

I don't see a noteworthy performance improvement on my phone, but I do see a noteworthy performance improvement when removing the `ECF_A16B16G16R16F` branch so it uses `ECF_A8R8G8B8` instead. The data I collected might not be valid though.

Reply:
thx for testing

> I do see a noteworthy performance improvement when removing the ECF_A16B16G16R16F branch so it uses ECF_A8R8G8B8 instead

OpenGL has a 10-bit color format and I have wondered before if we should maybe try using it, in case it's faster than 16-bit.
Not exactly sure what the higher precision is for anyway. It might be so bloom, tonemapping and light effects can work correctly? (see: HDR in video)